taca/backup/backup.py

```diff
@@ -0,0 +1,309 @@
+        """Check if a subprocess status is success and log error if failed"""
+        if status != 0:
+            self._clean_tmp_files(files_to_remove)
+            logger.error("Command '{}' failed with the error '{}'".format(" ".join(cmd),err_msg))
+            return False
+        return True
+
+    def _clean_tmp_files(self, files):
+        """Remove the file is exist"""
+        for fl in files:
+            if os.path.exists(fl):
+                os.remove(fl)
+
+    @classmethod
+    def encrypt_runs(cls, run, force):
+        """Encrypt the runs that have been collected"""
+        bk = cls(run)
+        bk.collect_runs(ext=".tar.gz")
+        logger.info("In total, found {} run(s) to be encrypted".format(len(bk.runs)))
+        for run in bk.runs:
+            run.flag = "{}.encrypting".format(run.name)
+            run.dst_key_encrypted = os.path.join(bk.keys_path, run.key_encrypted)
+            tmp_files = [run.zip_encrypted, run.key_encrypted, run.key, run.flag]
+            logger.info("Encryption of run {} is now started".format(run.name))
+            # Check if there is enough space
+            if not bk.avail_disk_space(path=run.path, run=run.name):
+                logger.error("There is no enough disk space for compression, kindly check and archive encrypted runs")
+                raise SystemExit
+            # Check if the run in demultiplexed
+            if not force and bk.check_demux:
+                if not bk.run_is_demuxed(run.name):
+                    continue
+            with filesystem.chdir(run.path):
+                # skip run if already ongoing
+                if os.path.exists(run.flag):
+                    logger.warn("Run {} is already being encrypted, so skipping now".format(run.name))
+                    continue
+                flag = open(run.flag, 'w').close()
+                # zip the run directory
+                if os.path.exists(run.zip):
+                    if os.path.isdir(run.name):
+                        logger.warn("Both run source and zipped archive exist for run {}, skipping run as precaution".format(run.name))
+                        bk._clean_tmp_files([run.flag])
+                        continue
+                    logger.info("Zipped archive already exist for run {}, so using it for encryption".format(run.name))
+                else:
+                    logger.info("Creating zipped archive for run {}".format(run.name))
+                    if bk._call_commands(cmd1="tar -cf - {}".format(run.name), cmd2="pigz --fast -c -",
+                                         out_file=run.zip, tmp_files=[run.zip, run.flag]):
+                        logger.info("Run {} was successfully compressed, so removing the run source directory".format(run.name))
+                        shutil.rmtree(run.name)
+                    else:
+                        logger.warn("Skipping run {} and moving on".format(run.name))
```

Comment:
we are here id the call returns false right? in this case it means that zipping did not work well. Maybe also in this case is worth sending us a mail and the log message should be more informative